shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxData.hs

```diff
@@ -431,9 +432,11 @@ pattern TxOut addr coin <-
 viewCompactTxOut :: forall crypto. Crypto crypto => TxOut crypto -> (Addr crypto, Coin)
 viewCompactTxOut (TxOutCompact bs c) = (addr, coin)
   where
-    addr = case deserialiseAddr (BSS.fromShort bs) of
-      Nothing -> panic "viewCompactTxOut: impossible"
+    addr = case deserializeShortAddr bs of -- Try to deserialize a Shelley style Addr directly from ShortByteString
       Just (a :: Addr crypto) -> a
+      Nothing -> case deserialiseAddr (BSS.fromShort bs) of -- It is a Byron Address, try the more expensive route.
+        Nothing -> panic "viewCompactTxOut: impossible"
+        Just (a :: Addr crypto) -> a
```

Comment:
Great! Is the goal to eventually let `deserializeShortAddr` support Byron addresses too?